web/src/filter.js

```diff
@@ -727,6 +732,11 @@ export class Filter {
                 }
                 sender = user.full_name;
             }
+
+            if (user && people.should_add_guest_user_indicator(user.user_id)) {
+                return $t({defaultMessage: "Messages sent by {sender} (guest)"}, {sender});
+            }
+
             return $t(
                 {defaultMessage: "Messages sent by {sender}"},
```

Comment:
I feel like this pattern feels very awkward/ugly -- ideally, we'd not create duplicates of strings like this for the guest case, but instead just have the `{sender}` part be substituted with a translation of `{name} (guest}`, like what you generate in https://github.com/zulip/zulip/pull/26766/files#r1328971956. Care may need to be taken to avoid double-escaping.

It's likely worth a thread in #translating about what to do here. But I'm thinking that for use cases that for whatever reason don't support styling the "guest" part differently via say including the template, or something fancier, we probably want to avoid needing a separate duplicate of a string just for this feature if at all possible.